### PR TITLE
Add contracts endpoint

### LINKED ISSUE #12: Add contracts endpoint by address
# What is needed?
Implement the contracts endpoint by id following the same structure than https://safe-transaction-mainnet.safe.global/#/contracts/contracts_retrieve with multichain support.

# Proposal multichain support
I would like to add the multichain support by default for this and other endpoints and filter by query param the chain that the client wants. 
For example:
`GET contracts/{address}/?chain_id=chain_id
`If chain_id is not used to filter the endpoint should return all the contract and abis for all the chains where the contract is supported. 
If chain_id is used to filter just return the corresponding chain
Even we could think about add the chain_id as an array parameter to get a group of chains.  

# Tasks
- [x] Add the necessary models (chains, Contracts, Abis... )
- [x] Add the endpoint contract by address
- [x] Add test for the implemented endpoint

### PR DESCRIPTION
**Pull request by moisses89**
# Description
- Add the contracts endpoint following the next schema
```
{
  "count": 0,
  "next": "string",
  "previous": "string",
  "results": [
    {
      "address": "string",
      "name": "string",
      "display_name": "string",
      "chain_id": 0,
      "project": {
        "description": "string",
        "logo_file": "string"
      },
      "abi": {
        "abi_json": [
          {}
        ],
        "abi_hash": "string"
      }
    }
  ]
}
```
- Implement pagination for this endpoint and for future endpoints
# Issue related
Close #12

### CODE REVIEW

File: app/routers/contracts.py [hunk 1]
```diff
@@ -1,21 +1,33 @@
...
+    pagination_params: PaginationParams = Depends(),
+    chain_ids: Annotated[list[int] | None, Query()] = None,
     session: AsyncSession = Depends(get_database_session),
-) -> Sequence[Contract]:
-    return await ContractService.get_all(session)
+) -> PaginatedResponse[Contract]:
+    if not fast_is_checksum_address(address):
+        raise HTTPException(status_code=400, detail="Address is not checksumed")
+
+    contracts_service = ContractService(request)
```
Comment (Uxio0) [inline app/routers/contracts.py:32]: Passing the `request` to the service and instantiating one service for every client does not sound right to me

File: app/routers/models.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    chain_id: int
+    project: ProjectPublic | None
+    abi: AbiPublic | None
+
+    class Config:
+        from_attributes = True
+
+    @field_validator("address")
+    @classmethod
+    def convert_to_checksum_address(cls, address):
```
Comment (Uxio0) [inline app/routers/models.py:52]: types

File: app/services/contract.py [hunk 3]
```diff
@@ -1,12 +1,18 @@
-from typing import Sequence
+from typing import Any, Sequence
+
+from fastapi import Request
 
 from sqlmodel.ext.asyncio.session import AsyncSession
 
 from app.datasources.db.models import Contract
+from app.services.pagination import GenericPagination, PaginatedResponse
```
Comment (Uxio0) [inline app/services/contract.py:8]: Relative imports

**Review by Uxio0 — CHANGES_REQUESTED** [review]

File: app/datasources/db/models.py [hunk 4]
```diff
@@ -38,16 +42,20 @@ class AbiSource(SqlQueryBase, SQLModel, table=True):
 
 class Abi(SqlQueryBase, SQLModel, table=True):
     id: int | None = Field(default=None, primary_key=True)
-    abi_hash: bytes = Field(nullable=False, index=True, unique=True)
+    abi_hash: bytes | None = Field(nullable=False, index=True, unique=True)
```
[the remark this answers is not in the document]
Comment (moisses89) [reply app/datasources/db/models.py:45]: It is the same as the primary key, setting `nullable=True` and `None` as secondary type let us to don´t pass abi_hash when you instantiate it. At this way we are able to do `Abi(abi_json=abi_json) `and calculate the `abi_hash` on create function.

**Review by Uxio0 — APPROVED** [review]

**Pull request merged**